climakitae/threshold_tools.py

```diff
@@ -558,32 +937,153 @@ def return_period(ams):
                 else:
                     return_period = -1.0 / (return_prob - 1.0)
                     return_period = round(return_period, 3)
-            except ValueError:
+            except (ValueError, ZeroDivisionError):
                 return_period = np.nan
 
-        return return_period
+            bootstrap_values = []
+            for _ in range(bootstrap_runs):
+                result = bootstrap(
+                    ams,
+                    distr=distr,
+                    data_variable=data_variable,
+                    arg_value=return_value,
+                )
+                bootstrap_values.append(result)
+
+            conf_int_array = np.percentile(
+                bootstrap_values, [conf_int_lower_bound, conf_int_upper_bound]
+            )
+            conf_int_lower_limit = conf_int_array[0]
+            conf_int_upper_limit = conf_int_array[1]
 
-    return_period = xr.apply_ufunc(
+        return return_period, conf_int_lower_limit, conf_int_upper_limit
+
+    return_period, conf_int_lower_limit, conf_int_upper_limit = xr.apply_ufunc(
         return_period,
         ams,
         input_core_dims=[["time"]],
         exclude_dims=set(("time",)),
-        output_core_dims=[[]]
+        output_core_dims=[[], [], []],
     )
 
     return_period = return_period.rename("return_period")
     new_ds = return_period.to_dataset()
+    new_ds["conf_int_lower_limit"] = conf_int_lower_limit
+    new_ds["conf_int_upper_limit"] = conf_int_upper_limit
 
     if multiple_points:
         new_ds = new_ds.unstack("allpoints")
 
     new_ds["return_period"].attrs[
         "return value"
     ] = "occurrence of a {} value event".format(str(return_value))
+    new_ds["conf_int_lower_limit"].attrs[
+        "confidence interval lower bound"
+    ] = "{}th percentile".format(str(conf_int_lower_bound))
+    new_ds["conf_int_upper_limit"].attrs[
+        "confidence interval upper bound"
+    ] = "{}th percentile".format(str(conf_int_upper_bound))
+
     new_ds.attrs = ams_attributes
     new_ds.attrs["distribution"] = "{}".format(str(distr))
 
     return new_ds
 
 
 #####################################################################
+# ARCHIVE
```

Comment:
Better to delete the old version. With version-control you can always get it back. Or just store it somewhere else easier for yourself just to keep the codebase cleaner.

Reply:
great catch! removed!